_posts/2022-03-13-software.md

```diff
@@ -0,0 +1,60 @@
+
+A software release covers all of the principles for open science and a brief decsription can be found below and longer decription can be found in the respective principle.
+  
+### [Workflow](https://climatecompatiblegrowth.github.io/guidelines/open%20science/2022/03/18/workflow.html)
+When developing a workflow for a software release it is important to develop clear objects and steps with tests so that bugs and issues can be identified and reported. This is to assure that the workflow is working as intended at the release and that bugs or issues can be fixed for next release. More advanced software development has tests inplace which are checked for each change in code to ensure consistency. One such integrated tool on Github is [Travis CI](https://docs.travis-ci.com/user/tutorial/) which builds the code and tests it with user defined tests. Travis CI has a payment plan, but for open source accounts it is free under travis-ci.com.
+  
+### [Version control](https://climatecompatiblegrowth.github.io/guidelines/)
+The software is first built using version control (such as [Git](https://git-scm.com/), Mercurial or Subversion). The version control tracks all the changes that you make and allows to also revert back to older versions. The version control system allows for collaborative work as each contributor/author can create an own "branch" and develop new code. The new code can be reviewed and compared to the other developments and "merged" to the main branch if the new developement is accepted. When the code is ready and tested it can be released.
+
+### [License](https://climatecompatiblegrowth.github.io/guidelines/practice/2022/03/01/licensing.html)
+When releasing a software it is important to think about under which license you want to release it under. There are many licenses, but they can be grouped into [three groups](https://ieeexplore.ieee.org/document/5370763):
+
+- Free-for-all licenses (permissive) This type of licenses require to acknowledge the original author(s) of the licensed work in any derivative work. However, it allows derivative works to be then published under a different type of license, and/or made proprietary.
+
+- Keep-open licenses This type of licenses require for the derivative work to be also released under open license. However, the derivative work (e.g. a software product) could then be included as one part of a larger proprietary work.
+
+- Share-alike (copyleft) licenses This type of licenses require that, if the licensed work is included in larger work or project, the entire work should be released open-source.
+
+For more information on open source licenses you can find more details [on Open Source initiative](https://opensource.org/licenses).
+
+### [Metadata](https://climatecompatiblegrowth.github.io/guidelines/)
+Metadata is "data about the data" meaning that it describes the context, quality and condition of a e.g. dataset.
```

Comment:
Main points here are that metadata:

- are machine readable, so the software release can be indexed (e.g. on PyPI, or Zenodo) [findable]
- follow a standard depending on where the software is released

Reply:
OK